### PR TITLE
Refactored the repo to align with the standard charm specs

### PR DESCRIPTION
**Pull request by agileshaw**
- Added Makefile, pyproject.toml, tox.ini, and rename.sh
- Removed .flake8 file in order to properly run pflake8
- Removed run_tests script because we now use tox for testing
- Re-structured the repo
- Update README and .gitignore

### CODE REVIEW

**Comment by agileshaw** [comment]
Unit tests: https://pastebin.ubuntu.com/p/d8YCnQVtmq/
Lint is being fixed in #3 
There's currently no functional tests available.

File: .gitignore [hunk 1]
```diff
@@ -1,8 +1,14 @@
...
+.build/
+.tox/
+.venv/
 build/
 env/
 venv/
 src/__pycache__/
-tests/__pycache__/
-apt-mirror.charm
+tests/unit/__pycache__/
```
Comment (rgildein) [inline .gitignore]: I think that we can use `__pycache__/` to ignore any `__pycache__` folder.

File: tests/unit/__init__.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+"""Unit test module."""
+
+import sys
+
+sys.path.append(".")
```
Comment (rgildein) [inline tests/unit/__init__.py]: Why we need this?

File: tests/unit/requirements.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+flake8
```
Comment (rgildein) [inline tests/unit/requirements.txt]: I do not think we need `flake8`  in unit tests.

File: tests/unit/test_charm.py [hunk 4]
```diff
@@ -10,7 +10,7 @@
 )
 from uuid import uuid4
 from ops.testing import Harness
-from charm import AptMirrorCharm
+from src.charm import AptMirrorCharm
```
Comment (rgildein) [inline tests/unit/test_charm.py]: I think we should use `from charm import AptMirrorCharm` and define proper `PYTHONPATH` in `tox.ini`.

File: tests/unit/test_charm.py [hunk 5]
```diff
@@ -62,7 +62,8 @@ class MockStat():
         harness.charm._stored.config = default_config
         action_event = Mock()
         harness.charm._on_update_status(action_event)
-        assert harness.model.unit.status == BlockedStatus('Last sync: Thu Jan  1 01:00:01 1970 '
+        print(harness.model.unit.status)
```
Comment (rgildein) [inline tests/unit/test_charm.py]: Did you remember to remove it?

File: tox.ini [hunk 6]
```diff
@@ -0,0 +1,86 @@
+# This file is managed by bootstack-charms-spec and should not be modified
```
Comment (rgildein) [inline tox.ini:2]: This is template for reactive charms, could we create one for `ops` and than use it here?
- wrong `PYTHONPATH`
- wrong coverage setting in unit tests

File: tox.ini [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+  SNAP_HTTPS_PROXY
+  OS_*
+
+[testenv:dev-environment]
+envdir = {toxinidir}/.venv
+deps =
+  pre-commit
+  {[testenv:lint]deps}
+  {[testenv:unit]deps}
+  ; {[testenv:func]deps} # temporarily comment it out until func tests are added
```
Comment (rgildein) [inline tox.ini]: I think we can add structure for func tests (folder, empty requirements, bundles folder, etc.) or use empty `deps` and warning message in `commands` . Because like this `dev-environment` is working but `tox -e func` will failed and `make funct` also.

**Review by rgildein — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (rgildein) [reply .gitignore]: Sorry it should be `**/__pycache__`

Comment (agileshaw) [reply .gitignore]: Done

File: tests/unit/requirements.txt (hunk 3 above)
Comment (agileshaw) [reply tests/unit/requirements.txt]: You are right, it is not needed

File: tests/unit/test_charm.py (hunk 4 above)
Comment (agileshaw) [reply tests/unit/test_charm.py]: Done

File: tests/unit/test_charm.py (hunk 5 above)
Comment (agileshaw) [reply tests/unit/test_charm.py]: Yes, this was used only for my local tests.

File: tox.ini (hunk 6 above)
Comment (agileshaw) [reply tox.ini:2]: I've applied the templates created in this [MP](https://code.launchpad.net/~rgildein/bootstack-charms-spec/+git/bootstack-charms-spec/+merge/429914)

File: tox.ini (hunk 7 above)
Comment (agileshaw) [reply tox.ini]: Done

**Comment by agileshaw** [comment]
Output of the placeholder functional tests: https://paste.ubuntu.com/p/8mKPSSCWJ8/

File: tests/unit/__init__.py [hunk 8]
```diff
@@ -0,0 +1 @@
+"""Unit test module."""
```
Comment (rgildein) [inline tests/unit/__init__.py]: I think that this whole file is not needed.

**Review by rgildein — APPROVED** [review]
LGTM, only one small nitpick

File: tests/unit/__init__.py (hunk 8 above)
Comment (agileshaw) [reply tests/unit/__init__.py]: File removed.

**Comment by agileshaw** [comment]
Unit test result after all the changes: https://pastebin.ubuntu.com/p/cRpMrydfwT/

**Review by peppepetra — APPROVED** [review]
LGTM

**Review by Pjack — APPROVED** [review]
